backend/src/main/scala/cromwell/backend/WorkflowBackendActor.scala

```diff
@@ -0,0 +1,137 @@
+package cromwell.backend
+
+import akka.actor.{Actor, ActorLogging}
+import akka.event.LoggingReceive
+import cromwell.backend.WorkflowBackendActor._
+import cromwell.backend.model.{JobDescriptor, WorkflowDescriptor}
+
+import scala.concurrent.Future
+import scala.util.{Success, Failure}
+
+object WorkflowBackendActor {
+
+  // Commands
+  sealed trait WorkflowBackendActorMessage
+  case class Execute(jobDescriptor: JobDescriptor) extends WorkflowBackendActorMessage
+  case object Abort extends WorkflowBackendActorMessage
+  case class Recover(jobDescriptor: JobDescriptor) extends WorkflowBackendActorMessage
+
+  // Events
+  sealed trait WorkflowBackendActorEvent
+
+  sealed trait ExecutionEvent extends WorkflowBackendActorEvent
+  case class ExecutionSucceeded(result: String) extends ExecutionEvent //Result TBD.
+  case class ExecutionFailed(throwable: Throwable) extends ExecutionEvent
+
+  sealed trait AbortEvent extends WorkflowBackendActorEvent
+  case object AbortSucceeded extends AbortEvent
+  case class AbortFailed(throwable: Throwable) extends AbortEvent
+
+  sealed trait RecoverEvent extends WorkflowBackendActorEvent
+  case class RecoverSucceeded(jobDescriptor: JobDescriptor) extends RecoverEvent //Same happens here.
+  case class RecoverFailed(throwable: Throwable) extends RecoverEvent
```

Comment:
as per the meeting on 4/6 recover is actually needed, although in the case of a local BE it might mean only execute, while in case of remote execution (e.g. JES) it has a more comprehensive behaviour